src/api/habits/adapters/json.ts

```diff
@@ -26,7 +30,75 @@ const jsonHabitServiceFactory = (): HabitService => {
         }))
       );
 
-  return { addHabit, getHabits };
+  const addEntry = (
+    goals: GoalWithHabitHistory[],
```

Comment:
ideally, you should not need to pass in the original data back in here in these calls - it makes it entirely possible to make an incorrect call and corrupt the original data store. Instead, pull from the original data store to ensure data integrity and keep the your code safe

Reply:
This makes sense. I realised that the factory itself doesn't hold any data, which is why I was relying on the component to store it. I was thinking that it was overhead to call `getHabits()` again to fetch the data, but now that I've been adding the feature to only pull in 7 days' worth of data into the FE component, it makes more sense that the factory itself should fetch the data and merge the changes.